src/Morphir/Elm/Backend/Codec/Gen.elm

```diff
@@ -0,0 +1,271 @@
+
+                Public (TypeAliasDefinition _ _) ->
+                    [ typeName |> toCamelCase |> VarPattern |> emptyRangeNode ]
+
+                _ ->
+                    []
+
+        funcExpr : Expression
+        funcExpr =
+            case typeDef of
+                Public (CustomTypeDefinition _ (Public constructors)) ->
+                    case constructors of
+                        [] ->
+                            Literal "Types without constructors are not supported"
+
+                        ctor :: [] ->
+                            ctor
+                                |> constructorToRecord e
+                                |> typeToEncoder [ Tuple.first ctor ]
+
+                        ctors ->
+                            let
+                                caseValExpr : Node Expression
+                                caseValExpr =
+                                    typeName
+                                        |> toCamelCase
+                                        |> FunctionOrValue []
+                                        |> emptyRangeNode
+
+                                cases : List ( Node Pattern, Node Expression )
+                                cases =
+                                    let
+                                        ctorToPatternExpr : Constructor extra -> ( Node Pattern, Node Expression )
+                                        ctorToPatternExpr ctor =
+                                            let
+                                                pattern : Pattern
+                                                pattern =
+                                                    deconsPattern (Tuple.first ctor) (Tuple.second ctor)
+
+                                                expr : Expression
+                                                expr =
+                                                    ctor
+                                                        |> constructorToRecord e
+                                                        |> typeToEncoder [ Tuple.first ctor ]
+                                                        |> customTypeTopExpr
+                                            in
+                                            ( emptyRangeNode pattern, emptyRangeNode expr )
+                                    in
+                                    ctors |> List.map ctorToPatternExpr
+                            in
+                            CaseExpression { expression = caseValExpr, cases = cases }
+
+                Public (TypeAliasDefinition _ tpe) ->
+                    typeToEncoder [ typeName ] tpe
+
+                _ ->
+                    Literal "Private types are not supported"
+    in
+    FunctionDeclaration function
+
+
+typeToEncoder : List Name -> Type extra -> Expression
+typeToEncoder varName tpe =
+    case tpe of
+        Reference fqName typeArgs _ ->
+            case fqName of
+                FQName _ _ [ "int" ] ->
```

Comment:
This is the kind of stuff that we should put into the SDK as a reusable pattern. It would be too much to do now but can you please add a TODO comment?

Reply:
You mean these should be parsed as data-structures represented in the SDK, right?

Reply:
First, it should be added as a pattern in the SDL. This would be `SDK.intPattern`. But this is longer term.